R/roxy_docsy.R

```diff
@@ -86,64 +116,164 @@
-        if (func_name %in% c("new", "initialize")){
-            next
-        }
+    
+    return(out)
+}
 
-        out <- paste0(
-            out
-            , .describe_function(aClass$public_methods[[func_name]], func_name)
-        )
+# [description] Generates vector of public fields documentation 
+# (both static fields and active bindings)
+.get_public_fields <- function(aClass) {
+    
+    # Intialize empty vector to append additional method descriptions to
+    out <- NULL
+    
+    # If this class has a parent class, do the parent first
+    if (!is.null(aClass$get_inherit())) {
+        # Append parent class public fields
+        out <- c(out, .get_public_fields(aClass$get_inherit()))
     }
-
-    out <- paste0(out, "}\n")
+    
+    # Add this class' static fields and active bindings
+    # Keep only unique
+    out <- unique(c(
+        out
+        , names(aClass$public_fields)
+        , names(aClass$active)
+    ))
+    
     return(out)
+    
 }
 
-# Document public fields (both static fields and active bindings)
-.public_member_block <- function(aClass){
+# [description] Generates block for public fields documentation 
+.public_field_block <- function(aClass){
 
     out <- paste0(
-        "@section Public Members:\n"
+        "@section Public Fields:\n"
         , "\\describe{\n"
     )
 
-    public_members <- c(
-        names(aClass$public_fields)
-        , names(aClass$active)
+    # Get names of public fields
+    public_fields <- .get_public_fields(aClass)
+    
+    # Construct document item for each field
+    items <- vapply(
+        public_fields
+        , FUN = function(x){
+            paste0(
+                "    \\item{\\bold{\\code{"
+                , x
+                , "}}}{: ~~DESCRIBE THIS FIELD~~}\n"
+            )
+        }
+        , FUN.VALUE = ""
     )
-
-    items <- sapply(public_members, function(x){
-        paste0(
-            "    \\item{\\bold{\\code{"
-            , x
-            , "}}}{: ~~DESCRIBE THIS FIELD~~}\n"
-        )
-    })
-
+    
+    # Collapse document items into block
     out <- paste0(
         out
         , paste0(items, collapse = "")
         , "}\n"
     )
+    
+    return(out)
+}
+
+# [description] Generate special methods block
+.special_methods_block <- function(aClass){
+    
+    descriptVec <- NULL
+    
+    # Iterate through this class' public methods
+    for (thisFuncName in SPECIAL_METHODS){
+        
+        # if not defined, skip it
+        if (!thisFuncName %in% names(aClass$public_methods)) {
+            next
+        }
+        
+        thisFuncDescript <- .describe_function(aClass$public_methods[[thisFuncName]], thisFuncName)
+        
+        # if the function is clone, put in standard documentation
+        if (thisFuncName == "clone") {
+            thisFuncDescript <- sub("~~DESCRIBE THE METHOD~~", CLONE_DESCRIBE, thisFuncDescript)
+            thisFuncDescript <- sub("~~DESCRIBE THIS PARAMETER ", CLONE_PARAM, thisFuncDescript)
```

Comment:
```suggestion
            thisFuncDescript <- sub("~~DESCRIBE THIS PARAMETER~~", CLONE_PARAM, thisFuncDescript)
```
When #11 gets merged.

Reply:
first time I've seen the new suggested-changes thing!

#11 is merged now, bw